cmd/image-builder/describeimg.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"slices"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/osbuild/images/pkg/blueprint"
+	"github.com/osbuild/images/pkg/distro"
+	"github.com/osbuild/images/pkg/imagefilter"
+)
+
+// Use yaml output by default because it is both nicely human and
+// machine readable and parts of our image defintions will be written
+// in yaml too.  This means this should be a possible input a
+// "flattended" image definiton.
+type describeImgYAML struct {
+	Distro string `yaml:"distro"`
+	Type   string `yaml:"type"`
+	Arch   string `yaml:"arch"`
+
+	// XXX: think about ordering (as this is what the user will see)
+	OsVersion string `yaml:"os_vesion"`
+
+	Bootmode        string `yaml:"bootmode"`
+	PartitionType   string `yaml:"partition_type"`
+	DefaultFilename string `yaml:"default_filename"`
+
+	// XXX: add pipelines here? maybe at least exports?
+	Packages *packagesYAML `yaml:"packages"`
```

Comment:
I think the pipeline should be added here, since otherwise the output won't be complete for installer images. It may make sense to do it in a smart way - include pipeline only if here are multiple pipelines with a package set defined? :thinking:

Reply:
Makes sense, I updated the commit to include the pipelines, I hope this captures what you wrote.